sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -179,45 +245,45 @@ private void setNodesFromNodeAccountIds(Client client) {
         }
     }
 
-    private CompletableFuture<O> executeAsync(Client client, int attempt, @Nullable Throwable lastException) {
-        if (attempt > maxAttempts) {
-            return CompletableFuture.<O>failedFuture(new Exception("Failed to get gRPC response within maximum retry count", lastException));
-        }
-
+    private Node getNodeForExecute(int attempt) {
         var node = nodes.get(nextNodeIndex);
         node.inUse();
 
         logger.trace("Sending request #{} to node {}: {}", attempt, node.accountId, this);
 
+        var nodeIsHealthy = node.isHealthy();
         if (!node.isHealthy()) {
             logger.warn("Using unhealthy node {}. Delaying attempt #{} for {} ms", node.accountId, attempt, node.delayUntil);
-
-            return Delayer.delayFor(node.delay(), client.executor)
-                .thenCompose((v) -> executeAsync(client, attempt + 1, lastException));
         }
 
-        var methodDescriptor = getMethodDescriptor();
-        var call = node.getChannel().newCall(methodDescriptor, CallOptions.DEFAULT);
+        return node;
+    }
+
+    private ProtoRequestT getRequestForExecute() {
         var request = makeRequest();
 
         // advance the internal index
         // non-free queries and transactions map to more than 1 actual transaction and this will cause
         // the next invocation of makeRequest to return the _next_ transaction
         advanceRequest();
 
-        var startAt = System.nanoTime();
+        return request;
+    }
 
-        return toCompletableFuture(ClientCalls.futureUnaryCall(call, request)).handle((response, error) -> {
-            var latency = (double) (System.nanoTime() - startAt) / 1000000000.0;
+    private CompletableFuture<O> executeAsync(Client client, int attempt, @Nullable Throwable lastException) {
+        if (attempt > maxAttempts) {
+            return CompletableFuture.<O>failedFuture(new Exception("Failed to get gRPC response within maximum retry count", lastException));
```

Comment:
This failure differs from the synchronous execute. I like the concrete `MaxAttemptsExceededException` more than the generic `Exception`.